compiler/rustc_resolve/src/late.rs

```diff
@@ -3214,7 +3240,12 @@ impl<'a: 'ast, 'b, 'ast> LateResolutionVisitor<'a, 'b, 'ast> {
             }
             ExprKind::Repeat(ref elem, ref ct) => {
                 self.visit_expr(elem);
-                self.resolve_anon_const(ct, IsRepeatExpr::Yes);
+                self.with_lifetime_rib(LifetimeRibKind::AnonConst, |this| {
+                    this.resolve_anon_const(ct, IsRepeatExpr::Yes)
+                });
+            }
+            ExprKind::ConstBlock(ref ct) => {
+                self.resolve_anon_const(ct, IsRepeatExpr::No);
```

Comment:
Why isn't this wrapped into `self.with_lifetime_rib(LifetimeRibKind::AnonConst...` like others?

Reply:
This is allowed. Should it be banned?
https://play.rust-lang.org/?version=nightly&mode=debug&edition=2021&gist=de2f1773546d4547c0a68bd01a77851f

Reply:
> Should it be banned?

I don't know.
My question was mostly about whether it preserves the existing behavior or not.
But it looks like `Scope::Body` is not used for `const { ... }` blocks, so it does preserve it.

Reply:
Inline const can reference non-static lifetimes (due to type inference it can't be feasibly banned in either HIR or AST level anyway), that's why the original code matches just `DefKind::AnonConst` but not `AnonConst | InlineConst`.